File: WifiManager/app/src/main/java/com/ua/wifimanager/MainActivity.java
package com.ua.wifimanager;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.net.DhcpInfo;
import android.net.wifi.WifiInfo;
import android.net.wifi.WifiManager;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        String service= Context.WIFI_SERVICE;
        WifiManager wifi= (WifiManager) getSystemService(service);
        WifiInfo info= wifi.getConnectionInfo();
        DhcpInfo dhcpInfo = wifi.getDhcpInfo();
        String bssid = info.getBSSID();
        if(bssid!=null){
            int strength = WifiManager.calculateSignalLevel(info.getRssi(),5);
            int speed = info.getLinkSpeed();
            String units=WifiInfo.LINK_SPEED_UNITS;
            int frequency = info.getFrequency();
            String ssid = info.getSSID().toString();
            int ipAddress = dhcpInfo.ipAddress;
            int gateway = dhcpInfo.gateway;
            int subnetMask = dhcpInfo.netmask;
            String ip = intToIp(ipAddress);
            String gatewayIp = intToIp(gateway);
            String subnetMaskIp = intToIp(subnetMask);
            boolean isHidden = info.getHiddenSSID();
            String cSummary= String.format("Connected to %s at %s%s. Strength %s/5",ssid,speed,units,strength);
            String textWifi = "SSID: " + ssid + "\n" +
                    "BSSID: " + bssid + "\n" +
                    "Frecuencia: " + frequency + " MHz\n" +
                    "Velocidad: " + speed + " Mbps\n" +
                    "Fuerza de señal: " + strength + " dBm\n" +
                    "¿Está oculta la red?: " + isHidden + "\n" +
                    "Dirección IP: " + ip + "\n" +
                    "Puerta de enlace: " + gatewayIp + "\n" +
                    "Máscara de subred: " + subnetMaskIp+"\n"+"\n"+cSummary;


            TextView text= findViewById(R.id.textWifi);
            text.setText(textWifi);

        }
    }
    private static String intToIp(int ipAddress) {
        return ((ipAddress & 0xFF) + "." + ((ipAddress >> 8) & 0xFF) + "." +
                ((ipAddress >> 16) & 0xFF) + "." + (ipAddress >> 24 & 0xFF));
    }
}

File: Acelerometro/app/src/main/java/es/ua/eps/acelerometro/MainActivity.kt
package es.ua.eps.acelerometro

import android.content.Context
import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener
import android.hardware.SensorManager
import android.hardware.SensorManager.SENSOR_DELAY_FASTEST
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class MainActivity : AppCompatActivity(),SensorEventListener {

    private var sensor: Sensor? = null
    private lateinit var sensorManager: SensorManager
    private lateinit var txt_X: TextView
    private lateinit var txt_Y:TextView
    private lateinit var txt_Z:TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        txt_X=findViewById(R.id.posX)
        txt_Y=findViewById(R.id.posY)
        txt_Z=findViewById(R.id.posZ)
        sensorManager = getSystemService(Context.SENSOR_SERVICE) as SensorManager
        if(sensorManager!=null){
            sensor=sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER)
        }
        if(sensor!=null){
            sensor = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER)?.also{
                sensorManager.registerListener(this,sensor,SENSOR_DELAY_FASTEST,SENSOR_DELAY_FASTEST)
            }
        }



    }

    override fun onSensorChanged(event: SensorEvent?) {
        if(event?.sensor?.type==Sensor.TYPE_ACCELEROMETER){
            val x = event.values[0]
            val y = event.values[1]
            val z = event.values[2]
            txt_X.text="X: ${x.toInt()}";
            txt_Y.text="Y: ${y.toInt()}";
            txt_Z.text="Z: ${z.toInt()}";

        }

    }

    override fun onAccuracyChanged(p0: Sensor?, p1: Int) {
        TODO("Not yet implemented")

    }

    override fun onDestroy() {
        sensorManager.unregisterListener(this)

        super.onDestroy()
    }
}